app/scripts/lib/notification-manager.js

```diff
@@ -27,12 +35,14 @@ class NotificationManager {
         extension.windows.update(popup.id, { focused: true })
       } else {
         // create new notification popup
-        extension.windows.create({
+        let popUp = extension.windows.create({
           url: 'notification.html',
           type: 'popup',
           width,
           height,
         })
+
+        popUp.then(this._onCreated)
```

Comment:
Should handle failure. At least a simple case like:
`.catch(log.error)`

Reply:
@danfinlay handled the error case and used saved id to close the popup. Thanks for pointing out. https://github.com/MetaMask/metamask-extension/pull/4597/commits/a22e5be585464ee7ede861e987ab4a1d7017065f